File: tests/fixtures/supabase/supabase-non-atomic-replace-pattern-broken/01-education-replace.ts
import { supabase } from '../client.js';

interface EducationRow {
  school: string;
  start_year: number;
}

export async function replaceEducation(userId: string, rows: EducationRow[]) {
  await supabase.from('education').delete().eq('user_id', userId);
  for (const row of rows) {
    await supabase.from('education').insert({ user_id: userId, ...row });
  }
}

File: tests/fixtures/supabase/supabase-non-atomic-replace-pattern-broken/01_education_replace.py
from lib.client import supabase


def replace_education(user_id: str, rows: list[dict]):
    try:
        supabase.table("education").delete().eq("user_id", user_id).execute()
        for row in rows:
            supabase.table("education").insert({"user_id": user_id, **row}).execute()
    except Exception:
        # Both steps share one swallowing handler — if insert fails after a
        # successful delete, the rows are gone with no error surfaced.
        pass
